homeassistant/components/stream/__init__.py

```diff
@@ -0,0 +1,134 @@
+"""
+Provide functionality to stream video source.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/stream/
+"""
+import logging
+import threading
+
+import voluptuous as vol
+
+from homeassistant.auth.util import generate_secret
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.loader import bind_hass
+
+from .const import DOMAIN, ATTR_STREAMS, ATTR_ENDPOINTS
+from .worker import stream_worker
+from .hls import async_setup_hls
+
+REQUIREMENTS = ['av==6.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({}),
+}, extra=vol.ALLOW_EXTRA)
+
+
+@bind_hass
+def request_stream(hass, stream_source, *, fmt='hls',
+                   keepalive=False, options=None):
```

Comment:
For the record, I'm fine with keeping `keepalive` here and have it part of the stream component. I just don't think that it makes sense as part of a play_stream service for camera component. If a component wants keepalive/preload, we need some other type of config I think, and in a future PR.